### PR TITLE
Update

### PR DESCRIPTION
**Pull request by neizvestnost**
New check

### CODE REVIEW

File: new/src/MainDiagramBlock/ConfigurationForTextMessage.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    },
+    {
+        src:fit,
+        altText:"FullScreen",
+        text:"Warrning! Your monthly traffic..."
+    },
+    {
+        src:fit,
+        altText:"FullScreen",
+        text:"Info! You have 24 unread mess..."
```
Comment (romko-kozak) [inline new/src/MainDiagramBlock/ConfigurationForTextMessage.js]: why did you put ... at the end of the message? Text overlapping should be covered with css

File: new/src/MainDiagramBlock/MiniDiagrams.js [hunk 2]
```diff
@@ -1,35 +1,25 @@
...
-        ]
-        return (
 
-            <div className="mini-diagrams">
-                {this.createAllMiniDiagrams()}
 
+        return (
 
+            <div className="mini-diagrams">
+                {miniDiagrams['diagrams']}
```
Comment (romko-kozak) [inline new/src/MainDiagramBlock/MiniDiagrams.js]: facepalm... what is that?

File: new/src/MainDiagramBlock/TextInMiniDiagrams.js [hunk 3]
```diff
@@ -4,14 +4,15 @@ import React, { Component } from 'react';
 class TextInMiniDiagrams extends Component{
     render() {
         const text = ["43% members","35% sales","21% dLoss"]
+        const diagramTextTag = {textForBlueDiagram:<p>{text[0]}</p>,textForGreenDiagram:<p>{text[1]}</p>,textForRedDiagram:<p>{text[2]}</p> }
```
Comment (romko-kozak) [inline new/src/MainDiagramBlock/TextInMiniDiagrams.js]: one object

File: new/src/MainDiagramBlock/TextMessage.js [hunk 4]
```diff
@@ -3,17 +3,18 @@ import React, { Component } from 'react';
 
 class TextMessage extends Component{
     render() {
+        var {src,text,altText} = this.props.img
```
Comment (romko-kozak) [inline new/src/MainDiagramBlock/TextMessage.js]: why var?

File: new/src/MainHeader/MainSearchBlock.js [hunk 5]
```diff
@@ -5,20 +5,31 @@ import search from '../../../new/src/Images/search.png';
...
 
+        state={
+            myValue:''
+        }
+
+    onChangeHandler=e=>{
+        this.setState({myValue:e.target.value})
+    }
+    onClickHandler =()=>{
+        alert(this.state.myValue);
```
Comment (romko-kozak) [inline new/src/MainHeader/MainSearchBlock.js]: why do we need this here?

you need to make a REAL filter.

File: new/src/MainHeader/MenuForSmallDevices.js [hunk 6]
```diff
@@ -114,42 +13,13 @@ class MenuForSmallDevices extends Component{
                     <input type="checkbox"/>
 
 
-                    <span></span>
-                    <span></span>
-                    <span></span>
+                    {burger}
+                    {burger}
```
Comment (romko-kozak) [inline new/src/MainHeader/MenuForSmallDevices.js]: prev version was better.

File: new/src/MainHeader/RightHeaderToogleBlock.js [hunk 7]
```diff
@@ -1,6 +1,14 @@
 import React, {Component} from "react";
 
 class RightHeaderToogleBlock extends Component{
+
+       state = {
+           flag: false
+       }
+
+    onChangeHandler=()=>{
+        this.setState({flag:!this.state.flag})
```
Comment (romko-kozak) [inline new/src/MainHeader/RightHeaderToogleBlock.js]: what does it do?
just changing state? then this is unneeded crap. you need to make real actions with the state, not just changing it.

File: new/src/MainDiagramBlock/MiniDiagrams.js (hunk 2 above)
Comment (neizvestnost) [reply new/src/MainDiagramBlock/MiniDiagrams.js]: I just misunderstood you in skype when you said this hash[hashname]

File: new/src/Aside/AsideProfileBlock.js [hunk 8]
```diff
@@ -6,18 +6,19 @@ import AsideMenu from "./AsideMenu";
...
-            h2:"Marian Lewi`s",
-            p:"Admin Manager"
+            nameOfProfileUser:"Marian Lewi`s",
+            workPosition:"Admin Manager"
         }
         return(
             <div className="aside-profile-block">
                 <div className="aside-profile-block-photo">
-                    <img src = {ava}/>
+                    <img src = {ava} alt={altText[0]}/>
```
[the remark this answers is not in the document]
Comment (neizvestnost) [reply new/src/Aside/AsideProfileBlock.js]: I don`t understand when we need to use array syntax, or just render text.

File: new/src/Aside/AsideMenu.js [hunk 9]
```diff
@@ -1,145 +1,33 @@
...
+  );
+  return (
+    <div className="menu">
+      <div className="accordion-menu">
+        {listOfArcticles.map(items => {
+          return (
+            <li className="item" id={items.idLi} key={items.idLi}>
+              <a href={items.linkAttr} className="btn">
+                <img src={items.imgLink} />
+                {items.headLineMenuText}{" "}
```
Comment (romko-kozak) [inline new/src/Aside/AsideMenu.js]: why do we need this string with space here? If you want to render string with variables, use string literals instead.

File: new/src/Aside/AsideMenu.js [hunk 10]
```diff
@@ -1,145 +1,33 @@
...
+                {items.idLi === "messages" && indicator}
```
Comment (romko-kozak) [inline new/src/Aside/AsideMenu.js]: why dont you put indicator as a property in an object?

File: new/src/Aside/AsideProfileBlockIcons.js [hunk 11]
```diff
@@ -1,32 +1,58 @@
...
+                                                     </div>)
+
+    return(
             <div className="aside-profile-block-icons">
                 {srcImages.map(items=>{
-                 return   <div className="aside-profile-block-icons-items"><img src={items}/></div>
+                 return   <div className="aside-profile-block-icons-items" key={srcImages.img}><img src={items.img} alt={items.altText}/></div>
                 })}
-                {blockIconsItems['user']}{blockIconsItems['message']}{blockIconsItems['view']}
+                {listOfProgress}
```
Comment (romko-kozak) [inline new/src/Aside/AsideProfileBlockIcons.js]: still need to be improved

File: new/src/Aside/AsideProfileBlock.js [hunk 12]
```diff
@@ -7,17 +7,17 @@ import AsideMenu from "./AsideMenu";
 class AsideProfileBlock extends Component{
     render(){
```
Comment (romko-kozak) [inline new/src/Aside/AsideProfileBlock.js]: we dont need this as a `Component`. There are no state interactions here, so we can make it stateless and export as a `const`

File: new/src/Diagrams/MainDiagrams.js [hunk 13]
```diff
@@ -1,59 +1,46 @@
...
+      },
+      {
+        strokeDasharray: "25 75",
+        centerText: "76",
+        diagramsStats: "New orders +25%"
+      }
+    ];
+
+   const listOfStats = strokes.map(items => {
+      return [
```
Comment (romko-kozak) [inline new/src/Diagrams/MainDiagrams.js]: why do you use an array to render that? we can use `Fragment` instead.

File: new/src/Diagrams/MainDiagrams.js [hunk 14]
```diff
@@ -1,59 +1,46 @@
...
-                    <SvgDiagramsText text={text[2]}/>
-
-                </div>
-            </div>
-        );
-    }
 
+class MainDiagrams extends Component {
+  render() {
+    const strokes = [
```
Comment (romko-kozak) [inline new/src/Diagrams/MainDiagrams.js]: no need to make it a `Component`

File: new/src/Diagrams/MainDiagramsDashboard.js [hunk 15]
```diff
@@ -1,18 +1,15 @@
...
-
-               <p>Dashboard</p>
-               <p>Home . Dashboard</p>
-           </div>
-       );
-   }
-}
-export default MainDiagramsDashboard;
\ No newline at end of file
+export default MainDiagramsDashboard;
```
Comment (romko-kozak) [inline new/src/Diagrams/MainDiagramsDashboard.js]: why `export default` ?

File: new/src/MainDiagramBlock/ConfigurationForTextMessage.js [hunk 16]
```diff
@@ -0,0 +1,25 @@
...
+  },
+  {
+    src: fit,
+    altText: "FullScreen",
+    text: "Warrning! Your monthly traffic..."
+  },
+  {
+    src: fit,
+    altText: "FullScreen",
+    text: "Info! You have 24 unread mess..."
```
Comment (romko-kozak) [inline new/src/MainDiagramBlock/ConfigurationForTextMessage.js]: still need to be improved.
overlapping should be made by css

File: new/src/MainDiagramBlock/InfoBlocks.js [hunk 17]
```diff
@@ -1,19 +1,18 @@
...
-
-        return (
-            <div className="first-user-block">
-                <TabContentImage tabContentImageSrc = {this.props.tabContentImageSrc}/>
-                <TextMessage img = {this.props.img}/>
-            </div>
-
-        );
-    }
+class InfoBlocks extends Component {
```
Comment (romko-kozak) [inline new/src/MainDiagramBlock/InfoBlocks.js]: no need in `Component`

File: new/src/MainDiagramBlock/MainDiagramBlock.js [hunk 18]
```diff
@@ -9,68 +9,51 @@ import InfoBlocks from "./InfoBlocks";
...
+      },
+      {
+        src: cancel
+      }
+    ];
+    const { mainImg } = this.props.imagesForHeader;
+    const listOfInfoBlocks = tabContentImageSrc.map((items, index) => {
+      return [
+        <InfoBlocks
+          key={items.src}
```
Comment (romko-kozak) [inline new/src/MainDiagramBlock/MainDiagramBlock.js]: key would be better to make of some string + index

File: new/src/MainDiagramBlock/MainDiagramBlock.js [hunk 19]
```diff
@@ -9,68 +9,51 @@ import InfoBlocks from "./InfoBlocks";
...
+          tabContentImageSrc={items}
+          img={img[index]}
```
Comment (romko-kozak) [inline new/src/MainDiagramBlock/MainDiagramBlock.js]: that's not a good solution to use an index as an identifier here.
you'd better to think about more flexible one.

File: new/src/MainDiagramBlock/MainDiagramBlock.js [hunk 20]
```diff
@@ -9,68 +9,51 @@ import InfoBlocks from "./InfoBlocks";
...
+      <BigDiagram />,
+      <MiniDiagrams />,
+      <TextInMiniDiagrams />
+    ];
+    return (
+      <div className="main-diagram-block">
+        <MainStatisticBlockHeader
+          imagesForHeader={this.props.imagesForHeader}
+        />
+        {mainImg === messages ? listOfInfoBlocks : listOfDiagramsBlocks}
```
Comment (romko-kozak) [inline new/src/MainDiagramBlock/MainDiagramBlock.js]: you should not make checks with the hardcoded data

File: new/src/MainDiagramBlock/MainDiagramBlock.js [hunk 21]
```diff
@@ -9,68 +9,51 @@ import InfoBlocks from "./InfoBlocks";
...
+      </div>
+    );
+  }
 }
```
Comment (romko-kozak) [inline new/src/MainDiagramBlock/MainDiagramBlock.js]: no need in default export

File: new/src/MainDiagramBlock/MiniDiagrams.js [hunk 22]
```diff
@@ -1,35 +1,14 @@
...
-
-            </div>
-        );
-    }
-
-}
-
-export default MiniDiagrams;
\ No newline at end of file
+export default MiniDiagrams;
```
Comment (romko-kozak) [inline new/src/MainDiagramBlock/MiniDiagrams.js]: same here

File: new/src/Aside/AsideMenu.js (hunk 9 above)
Comment (neizvestnost) [reply new/src/Aside/AsideMenu.js]: It`s not me, it`s prettier

File: new/src/Aside/AsideProfileBlockIcons.js [hunk 23]
```diff
@@ -1,32 +1,54 @@
...
-          message:      <div className="aside-profile-block-icons-items"><p>1095</p><p>+56</p></div>,
-          view:   <div className="aside-profile-block-icons-items"><p>438</p><p>+38</p></div>
-
+import React, { Component } from "react";
+
+import userCircle from "../../Images/user-male-circle-filled.png";
+import sentArrow from "../../Images/sent-filled.png";
+import invisibleFilled from "../../Images/invisible-filled.png";
+
+const AsideProfileBlockIcons = () => {
```
Comment (romko-kozak) [inline new/src/Aside/AsideProfileBlockIcons.js]: export const?

File: new/src/Diagrams/MainDiagrams.js [hunk 24]
```diff
@@ -1,59 +1,47 @@
...
+        centerText: "76",
+        diagramsStats: "New orders +25%"
+      }
+    ];
+
+    const listOfStats = strokes.map(items => {
+      return (
+        <React.Fragment>
+          <SVGDiagrams strokes={items} />
+          <SvgDiagramsText strokes={items} />
```
Comment (romko-kozak) [inline new/src/Diagrams/MainDiagrams.js]: it's enough to pass only property, that exist in concrete component

File: new/src/MainDiagramBlock/MiniDiagrams.js [hunk 25]
```diff
@@ -1,35 +1,14 @@
-import React, { Component } from 'react';
+import React, { Component } from "react";
```
Comment (romko-kozak) [inline new/src/MainDiagramBlock/MiniDiagrams.js]: Component is declared but never used

File: new/src/MainDiagramBlock/TextInMiniDiagrams.js [hunk 26]
```diff
@@ -1,17 +1,19 @@
-import React, { Component } from 'react';
+import React, { Component } from "react";
 
+const TextInMiniDiagrams = () => {
```
Comment (romko-kozak) [inline new/src/MainDiagramBlock/TextInMiniDiagrams.js]: export const instead of default export?

File: new/src/MainHeader/MenuForSmallDevices.js [hunk 27]
```diff
@@ -159,4 +27,4 @@ class MenuForSmallDevices extends Component{
 }
 
 
-export default  MenuForSmallDevices;
\ No newline at end of file
+export default  MenuForSmallDevices;
```
Comment (romko-kozak) [inline new/src/MainHeader/MenuForSmallDevices.js]: no need in class. make it stateless. no need in default export

File: new/src/MainMessageBlock/TabContentImage.js [hunk 28]
```diff
@@ -5,25 +5,26 @@ class TabContentImage extends Component{
...
 
                         </div>
         );}
 
     }
 }
 
-export default TabContentImage;
\ No newline at end of file
+export default TabContentImage;
```
Comment (romko-kozak) [inline new/src/MainMessageBlock/TabContentImage.js]: just export const

**Pull request merged**